### ISSUE TITLE
Collect official images for candidates

### ISSUE THREAD

**Issue by elinaru** [caciviclab/disclosure-frontend#5]
image size and format TBA

**Comment by bayreporta** [comment]
Kent Graham - Mayoral Candidate

![kent_graham](https://cloud.githubusercontent.com/assets/2280866/10203193/687b3412-6769-11e5-8b66-8d0794654806.png)

**Comment by bayreporta** [comment]
Francisco Herrara - Mayoral Candidate

![francisco_herrera](https://cloud.githubusercontent.com/assets/2280866/10203217/890d73c0-6769-11e5-927e-35edf6056a36.jpg)

**Comment by bayreporta** [comment]
Ed Lee - Mayoral Candidate

![mayor_ed_lee_closeup](https://cloud.githubusercontent.com/assets/2280866/10203242/ae6fbf4c-6769-11e5-9eb9-c071c54c6ea8.jpg)

**Comment by bayreporta** [comment]
Reed Martin - Mayoral Candidate

![reed_martin](https://cloud.githubusercontent.com/assets/2280866/10203255/c1c00e12-6769-11e5-9cde-6c9e9b5b5e71.png)

**Comment by bayreporta** [comment]
Stuart Schuffman - Mayoral Candidate

![stuart_schuffman](https://cloud.githubusercontent.com/assets/2280866/10203277/d4c28f3a-6769-11e5-98c6-a2a7aed69e48.jpg)

**Comment by bayreporta** [comment]
Amy Farah Weiss - Mayoral Candidate

![amy_farah_weiss](https://cloud.githubusercontent.com/assets/2280866/10203312/035699ae-676a-11e5-9da4-de9e1d7f67c8.jpg)

**Comment by bayreporta** [comment]
Aaron Peskin - D3 Supe

![aaron_peskin](https://cloud.githubusercontent.com/assets/2280866/10203345/38e46646-676a-11e5-9b59-06f073775a82.jpg)

**Comment by bayreporta** [comment]
Julie Christensen - D3 Supe

![julie_christensen](https://cloud.githubusercontent.com/assets/2280866/10203361/50cd633e-676a-11e5-9b2a-2d1980eb1015.gif)

**Comment by bayreporta** [comment]
Wilma Pang - D3 Supe (Photo from SF Examiner)

![pang_0916](https://cloud.githubusercontent.com/assets/2280866/10203423/b369e512-676a-11e5-9f9d-b5bf38f1cd27.jpg)

**Comment by elinaru** [comment]
Thanks, John. Once we determine the max size we'll need one the site, I'll resize, crop, and compress these images. 

That photo of Wilma Pang isn't the most flattering. It's not recent, but maybe we should use her last official campaigning photo? 

![2740782](https://cloud.githubusercontent.com/assets/6467815/10207351/960fc458-6780-11e5-8769-23ff3b23d699.jpg)

**Comment by bayreporta** [comment]
Sounds like a plan

**Comment by adborden** [comment]
FYI adding django models so that you'll be able to enter these  right from disclosure.caciviclab.org.. I'll keep you posted.

**Comment by bcipolli** [comment]
Now, for each Candidate, there is a `photo_url`:

http://admin.caciviclab.org/admin/office_election/candidate/73/
![image](https://cloud.githubusercontent.com/assets/4072455/12392155/b23d0bb4-bda1-11e5-98a0-07f8989d21a6.png)

Note: If we load this information manually now, we should also have a way to save it. Fixtures aren't a great choice at the moment, because the data in the database isn't clean. Perhaps writing a simple load/dump script that does loads/saves metadata by `name`, rather than `id`.

**Comment by bcipolli** [comment]
We should probably also be very careful about where we get images from, perhaps writing a set of rules for how we get them, so that people don't question the method and suggest some bias (e.g. is there an official campaign photo? or perhaps facebook photo, twitter photo, etc)

**Comment by mikeubell** [comment]
What is the difference between "ballot item" and "election office"?

**Comment by bcipolli** [comment]
`BallotItem` is a generalization, encompassing both `Referendum` and `OfficeElection`.

**Comment by adborden** [comment]
@bcipolli is the candidate photo upload fixed?

**Comment by bcipolli** [comment]
@adborden It is.

**Comment by jnmarcus** [comment]
I think we should offload this to [Cloudinary.com](http://cloudinary.com/). Unless we want to handle image optimization and caching ourselves, this will really slow down the frontend.  Do you guys think we'll be able to stay within[ the free plan](http://cloudinary.com/pricing)?

**Comment by jnmarcus** [comment]
Also, I think ^it can be integrated directly with Django

**Comment by bcipolli** [comment]
@jnmarcus I don't understand, sorry. Django has facilities for uploading & serving photos. Django & the browser communicate to cache images when appropriate. We have 5 lines of code to specify to Django where to store the images uploaded by back-end admins who define candidate information.

Is there anything left for Cloudinary to solve? Remember: I'm relatively new at modern front-end technologies, so I'm asking because I figure I'm not seeing the whole picture.

**Comment by sseiter** [comment]
Hello Jamie and Ben,

I'm not a front end developer either, but I feel like if we can handle this with Django, and incorporating new services that would be best.  I'm hesitant to integrate Cloudinary if we might ever need to scale up from the free plan.  And since it is my hope that we'll be able to scale rapidly.   

A quick note: I've met with Spike about how our team will make technical decisions, since it has been a point of contention.  I'll say a bit more about this in our meeting but in short, if we reach an impasse on a technical decision we will have a board made of reps from each brigade, and a technical advisor to help serve as a tiebreaker for technical issues. 

In short, do you think we can resolve this with Django, or do we need to hurry and onboard the technical advisor?

**Comment by adborden** [comment]
Closing this, we don't need the mayoral candidates right now.
